testsys/src/run_vmware.rs

```diff
@@ -0,0 +1,255 @@
+        let vm_resource_name = self
+            .vm_resource_name
+            .unwrap_or(format!("{}-instances", self.cluster_name));
+        let secret_map = btreemap! [ AWS_CREDENTIALS_SECRET_NAME.to_string() => self.aws_secret.clone(),
+        VSPHERE_CREDENTIALS_SECRET_NAME.to_string() => self.vsphere_secret,
+        WIREGUARD_SECRET_NAME.to_string() => self.wireguard_secret ];
+
+        let encoded_kubeconfig = base64::encode(
+            read_to_string(&self.target_cluster_kubeconfig_path).context(error::FileSnafu {
+                path: self.target_cluster_kubeconfig_path,
+            })?,
+        );
+
+        let vm_config = VSphereVmConfig {
+            ova_name: self.ova_name,
+            vm_count: self.vm_count,
+            tuf_repo: bottlerocket_agents::TufRepoConfig {
+                metadata_url: self.tuf_repo_metadata_url,
+                targets_url: self.tuf_repo_targets_url,
+            },
+            vcenter_host_url: self.host_url,
+            vcenter_datacenter: self.datacenter,
+            vcenter_datastore: self.datastore,
+            vcenter_network: self.network,
+            vcenter_resource_pool: self.resource_pool,
+            vcenter_workload_folder: self.workload_folder,
+            cluster: bottlerocket_agents::VSphereClusterInfo {
+                name: self.cluster_name,
+                control_plane_endpoint_ip: self.cluster_endpoint,
+                kubeconfig_base64: encoded_kubeconfig.clone(),
+            },
+        }
+        .into_map()
+        .context(error::ConfigMapSnafu)?;
+
+        let vm_resource = Resource {
+            metadata: ObjectMeta {
+                name: Some(vm_resource_name.clone()),
+                namespace: Some(NAMESPACE.into()),
+                ..Default::default()
+            },
+            spec: ResourceSpec {
+                depends_on: None,
+                agent: Agent {
+                    name: "vsphere-vm-provider".to_string(),
+                    image: self.vm_provider_image,
+                    pull_secret: self.vm_provider_pull_secret,
+                    keep_running: false,
+                    timeout: None,
+                    configuration: Some(vm_config),
+                    secrets: Some(secret_map.clone()),
+                    capabilities: Some(self.capabilities.clone()),
+                },
+                destruction_policy: DestructionPolicy::OnDeletion,
+            },
+            status: None,
+        };
+
+        let test = Test {
+            metadata: ObjectMeta {
+                name: Some(self.name.clone()),
+                namespace: Some(NAMESPACE.into()),
+                ..Default::default()
+            },
+            spec: TestSpec {
+                resources: vec![vm_resource_name.clone()],
+                depends_on: Default::default(),
+                agent: Agent {
+                    name: "sonobuoy-test-agent".to_string(),
```

Comment:
nit: maybe namespace this in a way that's specific to vmware
```suggestion
                    name: "vmware-sonobuoy-test-agent".to_string(),
```